### PR TITLE
plist: add plist support

### PR DESCRIPTION
**Pull request by GUIDINGLI**
## Summary

plist: add plist support
notifier: add BLOCKING_INIT_NOTIFIER_HEAD support

## Impact

notifier list plist

## Testing

VELA

### CODE REVIEW

File: drivers/cpufreq/cpufreq_performance.c [hunk 1]
```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ *  linux/drivers/cpufreq/cpufreq_performance.c
+ *
+ *  Copyright (C) 2002 - 2003 Dominik Brodowski <linux@brodo.de>
+ */
+
```
Comment (acassis) [inline drivers/cpufreq/cpufreq_performance.c]: @GUIDINGLI I think we cannot use this because GPL license is incompatible with Apache License.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by acassis** [comment]
@GUIDINGLI @xiaoxiang781216 these headers were copied from here: https://elixir.bootlin.com/linux/latest/source/include/linux/list.h

Copying GPL code to NuttX means that we will need to change NuttX license from Apache to GPL to comply with the license. Please be careful and *never* include Linux code inside NuttX, unless you contact the original author and he accept to submit is as Apache License to NuttX.

File: include/nuttx/list.h [hunk 2]
```diff
@@ -79,6 +79,15 @@
...
 
+/**
+ * list_entry_is_head - test if the entry points to the head of the list
+ * @pos:    the type * to cursor
+ * @head:   the head for your list.
+ * @member: the name of the list_head within the struct.
+ */
+#define list_entry_is_head(pos, head, member) \
+            (&pos->member == (head))
+
```
Comment (acassis) [inline include/nuttx/list.h:90]: Besides been copied from Linux, these comments are not following the NuttX coding style

**Review by davids5 — CHANGES_REQUESTED** [review]
@GUIDINGLI @xiaoxiang781216 - this is a copy of files from Linux with mild formatting changes. It can not come in.

**Comment by xiaoxiang781216** [comment]
> @GUIDINGLI @xiaoxiang781216 - this is a copy of files from Linux with mild formatting changes. It can not come in.

Ok, let's drop plist.h related change and rewrite a new one.

**Comment by jerpelea** [comment]
please do not copy GPL code !

**Comment by slorquet** [comment]
The idea that this pull request could have been accepted is maybe the more worrying point.

**Comment by slorquet** [comment]
Also what does "VELA" means in the testing section of the pull request initial text?

**Comment by pkarashchenko** [comment]
> Also what does "VELA" means in the testing section of the pull request initial text?
> 
> And what impact info can be understood from this description?

VELA is the name of Xiaomi OS based on NuttX I think.

**Comment by slorquet** [comment]
Are we supposed to know this, as upstream nuttx contributors?

**Pull request closed without merging**